pyvisa_sim/common.py

```diff
@@ -9,29 +9,51 @@
 
 """
 import logging
-from typing import Callable, Optional, Sequence
+from typing import Callable, Iterator, Optional, Sequence
 
 from pyvisa import logger
 
 logger = logging.LoggerAdapter(logger, {"backend": "sim"})  # type: ignore
 
 
-def iter_bytes(data: bytes, mask: Optional[int] = None, send_end: bool = False):
-    if send_end and mask is None:
-        raise ValueError("send_end requires a valid mask.")
+def iter_bytes(
+    data: bytes, data_bits: Optional[int] = None, send_end: bool = False
+) -> Iterator[bytes]:
+    """Clip values to the correct number of bits per byte.
 
-    if mask is None:
+    Serial communication may use from 5 to 8 bits.
+
+    Parameters
+    ----------
+    data : The data to clip as a byte string.
+    data_bits : How many bits per byte should be sent. Clip to this many bits.
+        For example: data_bits=5: 0xff (0b1111_1111) --> 0x1f (0b0001_1111)
+    send_end : If True, send the final byte unclipped (with all 8 bits).
```

Comment:
```diff
-(with all 8 bits)
+(with all `data_bits` bits)
```

Pending clarification of https://github.com/pyvisa/pyvisa-sim/pull/81#issuecomment-1526132661 - might be different.